engine/execution/provider/engine_test.go

```diff
@@ -71,6 +72,11 @@ func TestProviderEngine_onChunkDataRequest(t *testing.T) {
 		// submit using origin ID with invalid role
 		unittest.RequireCloseBefore(t, e.Ready(), 100*time.Millisecond, "could not start engine")
 		require.NoError(t, e.Process(channels.RequestChunks, originID, req))
+
+		require.Eventually(t, func() bool {
+			return requestQueue.Size() == uint(0) // ensuring all requests have been picked up from the queue.
+		}, 1*time.Second, 100*time.Millisecond)
```

Comment:
```suggestion
		}, 1*time.Second, 10*time.Millisecond)
```
Just guessing, but seems likely this will usually take <100ms

Reply:
https://github.com/onflow/flow-go/pull/2951/commits/63d9163669da376b331e728c89c949c497e9b719